src/ComunicaEngine.js

```diff
@@ -22,23 +23,38 @@ export default class ComunicaEngine {
    * Creates an asynchronous iterable of results for the given SPARQL query.
    */
   async* execute(sparql, source) {
-    if ((/^\s*(?:INSERT|DELETE)/i).test(sparql))
+    if ((/^\s*(?:INSERT|DELETE)/i).test(sparql)) {
       yield* this.executeUpdate(sparql, source);
-
-    // Load the sources if passed, the default sources otherwise
-    const sources = await (source ? this.parseSources(source) : this._sources);
-    if (sources.length !== 0) {
-      // Execute the query and yield the results
-      const queryResult = await this._engine.query(sparql, { sources, ...this._options });
-      yield* this.streamToAsyncIterable(queryResult.bindingsStream);
+    }
+    else {
+      // Load the sources if passed, the default sources otherwise
+      const sources = await (source ? this.parseSources(source) : this._sources);
+      if (sources.length !== 0) {
+        // Execute the query and yield the results
+        const queryResult = await this._engine.query(sparql, { sources, ...this._options });
+        yield* this.streamToAsyncIterable(queryResult.bindingsStream);
+      }
     }
   }
 
   /**
    * Creates an asynchronous iterable with the results of the SPARQL UPDATE query.
    */
   async* executeUpdate(sparql, source) {
-    throw new Error(`SPARQL UPDATE queries are unsupported, received: ${sparql}`);
+    // Load the sources if passed, the default sources otherwise
+    let sources = await (source ? this.parseSources(source) : this._destination);
+    if (sources.length !== 0) {
+      // Update queries can only handle a single source
+      if (sources.length > 1)
+        sources = [sources[0]];
```

Comment:
The silent failure… wouldn't that be a problem?

Reply:
Perhaps I should change the behavior to be:
If the destination parameter is used - exactly 1 source must be provided otherwise an error is thrown here
If the destination parameter is not used, use the sources - if no sources are provided, throw an error, otherwise use the first source?

Reply:
Yes, but instead of the last bit, error when no sources are given?